dot/network/connmgr_test.go

```diff
@@ -86,3 +86,34 @@ func TestProtectUnprotectPeer(t *testing.T) {
 	unprot = cm.unprotectedPeers([]peer.ID{p1, p2, p3, p4})
 	require.Equal(t, unprot, []peer.ID{p1, p2, p3, p4})
 }
+
+func TestPersistentPeers(t *testing.T) {
+	configA := &Config{
+		BasePath:    utils.NewTestBasePath(t, "node-a"),
+		Port:        7000,
+		RandSeed:    1,
+		NoBootstrap: true,
+		NoMDNS:      true,
+	}
+	nodeA := createTestService(t, configA)
+
+	addrs := nodeA.host.multiaddrs()
+	configB := &Config{
+		BasePath:        utils.NewTestBasePath(t, "node-b"),
+		Port:            7001,
+		RandSeed:        2,
+		NoMDNS:          true,
+		PersistentPeers: []string{addrs[0].String()},
+	}
+	nodeB := createTestService(t, configB)
+
+	// B should have connected to A during bootstrap
+	conns := nodeB.host.h.Network().ConnsToPeer(nodeA.host.id())
+	require.NotEqual(t, 0, len(conns))
```

Comment:
I don't understand this, shouldn't len(conns) == 1, since nodeB should be connected to nodeA?

Reply:
yeah, this just checks that it's greater than 0, there might be multiple conns open potentially